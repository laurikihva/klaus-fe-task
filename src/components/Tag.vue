<template>
  <div :class="computedClasses">
    <span class="text-xs font-medium">{{ label }}</span>
  </div>
</template>

<script lang="ts" setup>
import { toRefs, computed } from "vue";

export type TTagColor = "pink" | "purple" | "blue" | "yellow";
export interface ITagProps {
  color: TTagColor;
  label: string;
}

const props = defineProps<ITagProps>();
const { color } = toRefs(props);

const tagColors = computed(() => {
  return {
    pink: "bg-pink-100 text-pink-200",
    purple: "bg-purple-100 text-purple-200",
    blue: "bg-blue-200 text-blue-300",
    yellow: "bg-yellow-100 text-yellow-200",
  }[color.value];
});

const computedClasses = computed(() => {
  const base = "inline-flex items-center min-h-[24px] rounded px-2 py-0.5";

  return [base, tagColors.value];
});
</script>
