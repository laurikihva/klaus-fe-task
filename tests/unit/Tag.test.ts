import { mount } from "@vue/test-utils";

import Tag from "@/components/Tag.vue";

const wrapper = mount(Tag, {
  props: {
    label: "My tag",
    color: "pink",
  },
});

describe("Loads and displays Tag component", () => {
  it("renders correctly", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("contains label and base classes", () => {
    expect(wrapper.text()).toContain("My tag");

    expect(wrapper.classes("inline-flex")).toBe(true);
    expect(wrapper.classes("bg-pink-100")).toBe(true);
    expect(wrapper.classes("rounded")).toBe(true);
    expect(wrapper.classes("min-h-[24px]")).toBe(true);
  });
});
