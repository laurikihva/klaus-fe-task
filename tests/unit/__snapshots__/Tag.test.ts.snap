// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loads and displays Tag component > renders correctly 1`] = `"<div class="inline-flex items-center min-h-[24px] rounded px-2 py-0.5 bg-pink-100 text-pink-200"><span class="text-xs font-medium">My tag</span></div>"`;
